pyat/at/acceptance/momap_alternative.py

```diff
@@ -0,0 +1,402 @@
+"""momentum aperture alternative."""
+
+from __future__ import annotations
+
+import time
+from typing import Dict
+
+import numpy
+
+from ..lattice.lattice_object import Lattice
+
+__all__ = ["momaperture_project2start", "projectrefpts"]
+
+# 2024may29 oblanco at ALBA CELLS. First working version
+#           Based on the MATLAB implementation by Z.Marti at ALBA
+#           See https://github.com/atcollab/at/pull/773
+
+def momaperture_project2start(ring: Lattice, **kwargs: Dict[str, any]) -> numpy.ndarray:
+    """
+    :py:func:`momap_project2start` calculates the local momemtum aperture.
+
+    It is a binary search of the negative and positive energy thresholds
+    of stability around the closed orbit.
+
+    For a given energy offset the particles are first tracked from every
+    reference point to the end of the ring, and then, all or a set of particles
+    with different 6D coordinates are tracked together.  The surviving particles
+    continue the boundary search with a new energy step until the boundary is
+    found when the step limit or the  energy limit is reached.
+
+    Usage:
+      >>> momaperture_project2start(ring)
+
+    Parameters:
+      ring: list of elements
+
+    Keyword Arguments:
+      refpts: Selects the location of coordinates output.
+        See ":ref:`Selecting elements in a lattice <refpts>`"
+      nturns: number of turns to be tracked. Default 1000
+      dptol: energy offset resolution. Default 1e-4
+      euguess: maximum energy boundary. Default: the rf bucket height.
```

Comment:
Is it energy or momentum ?

Reply:
@lfarv , you are right in asking, it is momentum.

Reply:
Solved in the last commit
```bash
-      dptol: energy offset resolution. Default 1e-4
+      dptol: momentum offset resolution. Default 1e-4
```